src/scripts/Build-ModuleDoc.ps1

```diff
@@ -0,0 +1,210 @@
+        [string]
+        $Function
+    )
+
+    $help = Get-Help -Name $Function -Full
+    $excludeCommonParameters = [System.Management.Automation.PSCmdlet]::CommonParameters + [System.Management.Automation.PSCmdlet]::OptionalCommonParameters
+    $examples = @()
+    $parameters = @()
+    foreach ($parameter in $help.parameters.parameter.Where({$excludeCommonParameters -notcontains $_.Name}))
+    {
+        $parameters += @{
+            Name         = $parameter.name
+            Description  = $parameter.description.text | Select-Object -First 1
+            DefaultValue = $parameter.defaultValue
+            Required     = $parameter.required
+            Type         = $parameter.parameterValue
+        }
+    }
+
+    foreach($example in $help.examples.example)
+    {
+        $examples += @{
+            Id     = $example.title -replace '\D'
+            Remark = $example.remarks.text | Select-Object -First 1
+            Code   = $example.code
+        }
+    }
+
+    return @{
+        Description = $help.Description
+        Syntax      = ($help.syntax | Out-String) -replace '^[\r\n]+|\.|[\r\n]+$'
+        Synopsis    = $help.Synopsis
+        Examples    = $examples
+        Parameters  = $parameters
+    }
+}
+
+switch ($PSCmdlet.ParameterSetName)
+{
+    'ByPath'
+    {
+        if (-not [System.IO.Path]::IsPathRooted($Path))
+        {
+            $Path = Resolve-Path -Path $Path
+        }
+
+        Import-Module -FullyQualifiedName $Path -WarningAction 'SilentlyContinue'
+        $moduleName = [System.IO.Path]::GetFilenameWithoutExtension($Path)
+        $modulePath = Split-Path -Path $Path -Parent
+    }
+
+    'ByName'
+    {
+        if (-not (Get-Module -Name $Name -ListAvailable))
+        {
+            throw ('Module {0} not found.' -f $Name)
+        }
+
+        Import-Module -Name $Name -WarningAction 'SilentlyContinue'
+        $moduleName = $Name
+        $modulePath = Split-Path -Path (Get-Module -Name $moduleName).Path -Parent
+    }
+}
+
+$sb = [System.Text.StringBuilder]::new()
+[void]$sb.AppendLine('# {0} PowerShell Module' -f $moduleName).AppendLine()
+
+# Check for manifest, if it exists add Description and required modules
+$manifestPath = Get-ChildItem -Path (Split-Path -Path $modulePath -Parent) -Filter '*.psd1'
+if ($manifestPath)
+{
+    $manifest = Import-PowerShellDataFile -Path $manifestPath.FullName
+    [void]$sb.AppendLine($manifest.Description).Appendline()
+
+    if ($manifest.RequiredModules)
+    {
+        [void]$sb.AppendLine('## Required Modules').AppendLine()
+        [void]$sb.AppendLine('Name | Version')
+        [void]$sb.AppendLine('|:---:|:---:|')
+    
+        foreach ($requiredModule in $manifest.RequiredModules)
+        {
+            [void]$sb.AppendLine("$($requiredModule.ModuleName) | $($requiredModule.ModuleVersion)")
+        }
+
+        [void]$sb.AppendLine()
+    }        
+}
+
+# Parse functions
+[void]$sb.AppendLine('## Functions').AppendLine()
```

Comment:
nit: Should we use the term "commands" to generalize and avoid the difference between a cmdlet and a function, which doesn't matter to end users?
```suggestion
[void]$sb.AppendLine('## Commands').AppendLine()
```